### PR TITLE
Added API stubs necessary for the next task.

### PR DESCRIPTION
**Pull request by tommyp101**

### CODE REVIEW

**Comment by akdombrowski** [comment]
was that me that accidentally installed with yarn?
i use it for my projects and it's a habit

File: src/app/api/clients/addClientResponses.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * API Stub to Add Client Responses to Questions
+ * 
+ * @param {string} clientId - The email or phone number of the client submitting the response.
```
Comment (akdombrowski) [inline src/app/api/clients/addClientResponses.ts:4]: if it's one of either email or phone, I would just make 2 optional params and make sure one exists. it'll be easier if you know a variable is the same "type" to work with it

Comment (akdombrowski) [reply src/app/api/clients/addClientResponses.ts:4]: and reserve id for some kind of guuid-like string

File: src/app/api/clients/addClientResponses.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * @param {array} clientResponses - An array of objects, each containing a questionId and the client's answer.
+ * @returns {Promise} - A promise that resolves to an object containing:
+ *                      - status: boolean indicating if the operation was successful
+ *                      - message: a string message, either success or error message
+ */
+
+export default async function addClientResponses(clientId, clientResponses) {
```
Comment (akdombrowski) [inline src/app/api/clients/addClientResponses.ts:11]: it's not a hard rule, but I've found a good practice in general is to make the param and object by default. If you add another key to the object later to take in another param, it won't break things vs adding another one here

File: src/app/api/clients/addClientResponses.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return fetch('https://api.example.com/add-client-responses', {
```
Comment (akdombrowski) [inline src/app/api/clients/addClientResponses.ts:12]: api.example.com?

File: src/app/api/coaches/getClientResponses.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ * API Stub to Retrieve User's Answers
+ * 
+ * @param {string} userId - The unique identifier for the user (could be an email or user ID).
+ * @returns {Promise} - A promise that resolves to an object containing:
+ *                      - status: boolean indicating if the operation was successful
+ *                      - message: a string message, either success or error message
+ *                      - answers: an array of objects detailing each question and the user's response
+ */
+
+export default async function getUserAnswers(userId) {
```
Comment (akdombrowski) [inline src/app/api/coaches/getClientResponses.ts:11]: consistent naming with file

File: src/app/api/coaches/getPublishStatus.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * API Stub to Check Coach Questions Status
+ * 
+ * @param {string} coachEmail - The email of the coach to check, this is the primary key.
+ * @returns {Promise} - A promise that resolves to an object containing:
+ *                      - status: boolean indicating if the operation was successful
+ *                      - message: a string message, either success or error message
+ *                      - questionStatus: boolean indicating the coach's question status (true/false)
+ */
+
+export default async function getCoachQuestionStatus(coachEmail) {
```
Comment (akdombrowski) [inline src/app/api/coaches/getPublishStatus.ts:11]: same. consistent naming

File: src/app/api/fetchQuestions.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ * API Stub to Fetch Questions Assigned to a Coach
+ * 
+ * @param {string} coachEmail - The email of the coach to retrieve questions for.
+ * @returns {Promise} - A promise that resolves to an object containing:
+ *                      - status: boolean indicating if the operation was successful
+ *                      - message: a string message, either success or error message
+ *                      - questions: an array of objects with questionId, questionText, and answerType.
+ */
+
+export default async function fetchCoachQuestions(coachEmail) {
```
Comment (akdombrowski) [inline src/app/api/fetchQuestions.ts:11]: naming

File: tsconfig.json
Comment (akdombrowski) [inline tsconfig.json:1]: was this intentional?

File: src/app/onboarding/[linkId]/page.tsx
Comment (akdombrowski) [inline src/app/onboarding/[linkId]/page.tsx:1]: why is this its own page?

File: src/app/api/coaches/getPublishStatus.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    return fetch('https://api.example.com/check-question-status', {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        // TODO:  This would either be in the header/query of the request, since it's a GET request, remove if necessary.
+        body: JSON.stringify({ email: coachEmail })
```
Comment (akdombrowski) [inline src/app/api/coaches/getPublishStatus.ts:18]: I don't think you need an object as the body + it's a GET anyways.

File: src/app/api/coaches/getClientResponses.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    return fetch(`https://api.example.com/get-user-answers/${userId}`, {
```
Comment (akdombrowski) [inline src/app/api/coaches/getClientResponses.ts:12]: if this is an email with `@` in it, will it work?

File: src/app/api/coaches/getPublishStatus.ts
Comment (akdombrowski) [inline src/app/api/coaches/getPublishStatus.ts:1]: is there a setter along with this?

File: src/actions/link/generateLink.ts [hunk 9]
```diff
@@ -5,6 +5,7 @@ import { nanoid } from 'nanoid'; // Or any other method for generating unique ID
 export default async function generateLink() {
     const linkId = nanoid(); // Generates a unique ID
     // Save linkId and any associated data to your database
+    
     const baseUrl = process.env.ORIGIN
     const newLink = `${baseUrl}/onboarding/${linkId}`;
```
Comment (akdombrowski) [inline src/actions/link/generateLink.ts:10]: we can change the name later, not a big deal, but these questions don't necessarily have to be just for onboarding

File: src/actions/link/generateLink.ts [hunk 10]
```diff
@@ -5,6 +5,7 @@ import { nanoid } from 'nanoid'; // Or any other method for generating unique ID
 export default async function generateLink() {
     const linkId = nanoid(); // Generates a unique ID
```
Comment (akdombrowski) [inline src/actions/link/generateLink.ts:6]: we agreed to use coach's name

**Review by akdombrowski — CHANGES_REQUESTED** [review]

**Comment by tommyp101** [comment]
> was that me that accidentally installed with yarn? i use it for my projects and it's a habit

I'm not sure but I saw the yarn lock when I pulled, so I yarn install'd and it worked, just didn't deploy properly.

**Pull request merged**